com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -92,43 +89,38 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
         return StepPhase.POLL;
     }
 
-    private ApplicationResources getApplicationResources(ExecutionWrapper execution, String appArchiveId, String fileName)
-        throws FileStorageException {
-        ApplicationResources applicationResources = new ApplicationResources();
+    private String getNewApplicationDigest(ExecutionWrapper execution, String appArchiveId, String fileName) throws FileStorageException {
         DelegateExecution context = execution.getContext();
+        StringBuilder digestStringBuilder = new StringBuilder();

```

Comment:
Why StringBuilder is used?

Reply:
It's a hack to pass the string value out of the FileDownloadProcessor since it can't change the reference to an outside variable.